docs/modules/ROOT/pages/how-tos/release-process.adoc

```diff
@@ -25,10 +27,60 @@ git push -u
 
 . Check and update xref:tutorials/getting-started.adoc[Getting Started guide] and https://syn.tools/tuto/index.html[Tutorial: Writing your First Commodore Component] (These guides need to be up-to-date and work 100% all the time)
 
-== Additional work for major releases
+=== Additional work for major releases
 
 If a "major" release is being done, the following additional work needs to be done:
 
 * Update the https://github.com/projectsyn/documentation/blob/master/docs/modules/ROOT/pages/roadmap.adoc[Roadmap]
 * Write a blogpost and spread it on social media
 * Close the associated GitHub project and internal Jira project (if applicable)
+
+== Commodore Components
+
+Commodore Components are to be tagged with SemVer with `v<major>.<minor>.<patch>[-rc<rc>]` pattern (optionally includes Release candidates tagging).
+
+.Version examples
+====
+* `v1.0.0`
+* `v1.2.3-rc1`
+====
+
+[NOTE]
+====
+Generally, using `v0.x` versions are discouraged.
+After the first initial implementation, users are requested to directly release `v1.0.0`.
+Use release candidates suffix (for example `-rc1`) if "beta" versions are really needed.
+====
+
+The changelog is generated via GitHub Actions and is based on labelling PullRequests.
+The following GitHub labels require certain SemVer version increments:
+
+|===
+| Label | Increment
+
+| `bug`, `fix`, `change`, `dependency`
+| Patch
+
+| `enhancement`
+| Minor
+
+| `breaking`
+| Major
+|===
+
+The release process is relatively simple:
+
+. Identify which type of pull requests were merged since last release (fixes, enhancements or breaking changes) to decide on the next SemVer version number.
+. Create and push a new Git tag on the default branch.
+. If there are no breaking changes since last release, bump the version number of the released component in the Commodore defaults repository.
+
+NOTE: If there _are_ breaking changes, you can only bump the version in the defaults once all clusters have been upgraded to the new version.
```

Comment:
This note does not necessarily apply to Project Syn in general, and is much more an artifact of how we use Project Syn at the moment.

Additionally, it might be worth to outline a rough set of steps for rolling out breaking changes . There's two broad categories of such changes:

* Breaking changes which can be automatically rolled out with appropriate upgrade jobs in the component (e.g. https://github.com/projectsyn/component-cert-manager/releases/tag/v2.0.0)
* Breaking changes which require changes in the config hierarchy (e.g. https://github.com/appuio/component-openshift4-authentication/releases/tag/v2.0.0)

For the first category, it's generally sufficient to upgrade the component version.
For the second category, each instance of the changed configuration parameters needs to be updated before updating the component version to ensure cluster catalogs still compile with the new component version.

Reply:
I've tried to capture your thoughts in the latest commits